src/GafferScene/Rename.cpp

```diff
@@ -276,12 +342,70 @@ Gaffer::ValuePlug::CachePolicy Rename::computeCachePolicy( const Gaffer::ValuePl
 	return FilteredSceneProcessor::computeCachePolicy( output );
 }
 
+bool Rename::affectsOutputName( const Gaffer::Plug *input ) const
+{
+	return
+		input == namePlug() ||
+		input == deletePrefixPlug() ||
+		input == deleteSuffixPlug() ||
+		input == findPlug() ||
+		input == replacePlug() ||
+		input == addPrefixPlug() ||
+		input == addSuffixPlug()
+	;
+}
+
+void Rename::hashOutputName( IECore::MurmurHash &h ) const
+{
+	namePlug()->hash( h );
+	deletePrefixPlug()->hash( h );
+	deleteSuffixPlug()->hash( h );
+	findPlug()->hash( h );
+	replacePlug()->hash( h );
+	addPrefixPlug()->hash( h );
+	addSuffixPlug()->hash( h );
+}
+
+std::string Rename::outputName( IECore::InternedString inputName ) const
+{
```

Comment:
I think the ideal way of using this for ad-hoc renaming (rather than batch renaming where you can do it all with a regex) will be to connect a spreadsheet, where keeping `scene:path` is necessary. Because of the cost of name maps and set processing, I think there's a lot to be said for allowing the user to achieve all their possible renames with a single node-plus-spreadsheet rather than pushing them to use multiple nodes.

I hope that this node is only for when you don't have all your pipeline ducks in a row, and you're wrangling some temporarily bad names from the FX department, rather than some large-scale asset-managed renaming.

Reply:
I'm not sure I understand this.  I had said namePlug needs scene:path, but wasn't sure about the other plugs.  You suggested that users would use a spreadsheet ... are you picturing deleteSuffix being driven by a spreadsheet driven by scene:path?  If it was directly using scene:path, that doesn't seem to make sense, you'd probably just set the name if you know which path you're dealing with ... but I guess you can use globs, to do something like remove a suffix from all locations under "/characters/..."?